payment/src/main/java/org/killbill/billing/payment/api/DefaultPayment.java

```diff
@@ -93,29 +92,24 @@ public DefaultPayment(final UUID id, @Nullable final DateTime createdDate, @Null
-                                                                                                                                         }
-                                                                                                                                     }), null);
+        PaymentTransaction transactionToUseForCurrency = transactions.stream()
+                .filter(transaction -> (transaction.getTransactionType() == TransactionType.AUTHORIZE ||
+                                        transaction.getTransactionType() == TransactionType.PURCHASE ||
+                                        transaction.getTransactionType() == TransactionType.CREDIT) &&
+                                      (TransactionStatus.SUCCESS.equals(transaction.getTransactionStatus()) ||
+                                       TransactionStatus.PENDING.equals(transaction.getTransactionStatus())))
+                .findFirst().orElse(null);
+
         if (transactionToUseForCurrency == null) {
             // No successful one, take the last non-successful one then
-            transactionToUseForCurrency = Iterables.<PaymentTransaction>getLast(Iterables.<PaymentTransaction>filter(transactions,
-                                                                                                                     new Predicate<PaymentTransaction>() {
-                                                                                                                         @Override
-                                                                                                                         public boolean apply(final PaymentTransaction transaction) {
-                                                                                                                             return transaction.getTransactionType() == TransactionType.AUTHORIZE ||
-                                                                                                                                    transaction.getTransactionType() == TransactionType.PURCHASE ||
-                                                                                                                                    transaction.getTransactionType() == TransactionType.CREDIT;
-                                                                                                                         }
-                                                                                                                     }), null);
+            transactionToUseForCurrency = transactions
+                    .stream()
+                    .filter(transaction -> transaction.getTransactionType() == TransactionType.AUTHORIZE ||
+                                           transaction.getTransactionType() == TransactionType.PURCHASE ||
+                                           transaction.getTransactionType() == TransactionType.CREDIT)
+                    .findFirst().orElse(null);
```

Comment:
Shouldn't this be something like `findLast()` instead (`getLast` was used before)?